### ISSUE TITLE
Extending FrameProcessor.java causing Memory Leak in  Android.

### ISSUE THREAD

**Issue by madhaviKumari** [google/mediapipe#252]
I want to create my own ```CustomFrameProcessor``` so I extended from ```com.google.mediapipe.components.FrameProcessor``` , When I am using this ```CustomFrameProcessor``` object in TextureConverter class it is causing a memory leak. 
Note: Even I am not overriding existing ```onNewFrame(final TextureFrame frame)``` calling as it is from subclass instance still memory usage is monotonically increasing. 
Another point is by default ```FrameProcessor```class  have all members ```private``` so I modified them to ```protected``` for our need. So my question is why it is behaving like this? Can I not build my ```frameProcessor``` on the top of the existing ```FrameProcessor```.

**Comment by eknight7** [comment]
- Can you share some more details about the memory leak you are observing?
- What is the functionality you are trying to add in your `CustomFrameProcessor` ? Is it possible you share your code?

**Comment by madhaviKumari** [comment]
* Hi @eknight7  I could not find out the exact place where memory leak is happening but while using the object of ```FrameProcessor``` 's subclass then memory utilization is continuously increasing and app is crashing after few seconds. The same method call with ```FrameProcessor``` object is not causing this issue. 

* I want to pass two Frame object at a time to mediapipegraph that's why I am Extending ```FrameProcessor``` 

Following is code I implemented, For simplicity, I am sharing minimal code & problem is coming with this piece of code as well.

```java
public class CustomFrameProcessor extends FrameProcessor {
    private String bGVideoInputStream;

    /**
     * Constructor.
     *
     * @param context             an Android {@link Context}.
     * @param parentNativeContext a native handle to a GL context. The GL context(s) used by the
     *                            calculators in the graph will join the parent context's sharegroup, so that textures
     *                            generated by the calculators are available in the parent context, and vice versa.
     * @param graphName           the name of the file containing the binary representation of the graph.
     * @param inputStream         the graph input stream that will receive input video frames to be segmented.
     * @param outputStream        the output stream from which output frames will be produced.
     * @param bGVideoInputStream  the graph input stream that will receive background frames to be overlay.
     */
    public CustomFrameProcessor(Context context, long parentNativeContext, String graphName, String inputStream, String outputStream, String bGVideoInputStream) {
        super(context, parentNativeContext, graphName, inputStream, outputStream);
        this.bGVideoInputStream = bGVideoInputStream;
    }




    public void onNewFrame(AppTextureFrame foreground, AppTextureFrame background) {
        if (Log.isLoggable(TAG, Log.VERBOSE)) {
            Log.v(
                    TAG,
                    String.format(
                            "Input tex: %d width: %d height: %d",
                            foreground.getTextureName(), foreground.getWidth(), foreground.getHeight()));
        }

        if (!maybeAcceptNewFrame()) {
            foreground.release();
            background.release();
            return;
        }
        if (addFrameListener != null) {
            addFrameListener.onWillAddFrame(foreground.getTimestamp());
            addFrameListener.onWillAddFrame(background.getTimestamp());
        }

        Packet imagePacket = packetCreator.createGpuBuffer(foreground);
        Packet bgPacket = packetCreator.createGpuBuffer(background);
        try {
            // addConsumablePacketToInputStream allows the graph to take exclusive ownership of the
            // packet, which may allow for more memory optimizations.
            mediapipeGraph.addConsumablePacketToInputStream(
                    videoInputStream , imagePacket, foreground.getTimestamp());
            if(bgPacket!=null)
                mediapipeGraph.addConsumablePacketToInputStream(bGVideoInputStream,bgPacket,background.getTimestamp());
        } catch (MediaPipeException e) {
            Log.e(TAG, "Mediapipe error: ", e);
        }
        imagePacket.release();
        if(bgPacket!=null){
            bgPacket.release();
        }
    }
}


````

Note: If I am putting this newly implemented method in ```FrameProcessor``` class & using this then memory utilization is not growing and is working fine.

**Comment by eknight7** [comment]
Hi, I have couple follow-up questions:

1. Can you share which device you are using, and how are the two cameras connected?
2. Are you seeing this memory error on your own application, or are you trying to run one of the example applications such as face detection?
3. Can you share some memory logs? I am trying to reproduce this issue.

**Comment by eknight7** [comment]
@madhaviKumari gentle ping. If you were able to resolve this issue, please share what you did as well. Thanks!

**Comment by madhaviKumari** [comment]
Hey apology for late response, 
I tried on **Honor_Play** and  **VIVO 1609**, I am not using two cameras, One frame is coming from device Camera and second frame from MP4 video (Using ```SurfaceTexture```` same way like Camera frame is being used).
I am using ```Hair_Segmentation example```  with my own tflite_model file.

I had moved this method ``` onNewFrame(AppTextureFrame foreground, AppTextureFrame background) ```` from subclass to ```FrameProcessor``` itself then memory utilization was not increasing. 
I will share memory logs as well.
